### PR TITLE
Update API with some stubs that match the sketch

### PR DESCRIPTION
**Pull request by bowei**
- Update the API with some stubs to match the sketch.
- Some cleanup related to testing the CRDs using kind and kubebuilder.
- Minor updates to the documentation.

### CODE REVIEW

**Comment by bowei** [comment]
cc: @robscott

**Comment by bowei** [comment]
Rob -- looks like you need to be added to the org.

File: api/v1alpha1/gateway_types.go [hunk 1]
```diff
@@ -64,22 +61,56 @@ type GatewaySpec struct {
...
 
 const (
-	IPAddressType    = "IPAddress"
-	NamedAddressType = "NamedAddress"
+	// IPAddress is an address that is an IP address.
+	//
+	// Support: Extended.
+	IPAddress = "IPAddress"
+	// NamedAddress is an address selected by name. The interpretation of
+	// the name is depenedent on the controller.
```
Comment (howardjohn) [inline api/v1alpha1/gateway_types.go]: I am a bit confused what this would be. A hostname like `foo.com`?

Comment (bowei) [reply api/v1alpha1/gateway_types.go]: Idea is more like `us-central1/my-ip-address`
We are going to go over the sketch + API types in detail, the current status of the types is to mostly just have some CRDs to test out the kubebuilder infrastructure/test/CI infrastructure.

Comment (bowei) [reply api/v1alpha1/gateway_types.go]: Keep your comment in mind when we go over the gateway APIs :-)

**Pull request merged**